webapp/channels/src/utils/utils.tsx

```diff
@@ -212,6 +212,43 @@ const removeQuerystringOrHash = (extin: string): string => {
 };
 
 export const getFileType = (extin: string): typeof FileTypes[keyof typeof FileTypes] => {
+    // Handle null or undefined input
+    if (!extin) {
+        return FileTypes.OTHER;
+    }
+
+    // Special handling for image proxy URLs
+    // Check for various forms of image proxy URLs
+    if (extin.includes('/api/v4/image') &&
+        (extin.includes('?url=') || extin.includes('&url='))) {
+        return FileTypes.IMAGE;
+    }
+
+    // Check for image file extensions in the URL path
+    try {
+        // Try to parse as a URL - this will validate if it's a proper URL
+        const url = new URL(extin);
```

Comment:
But what if the URL does not contain any extension, which is of type `IMAGE_TYPE`, for example 

```
![Javatpoint](https://media-exp1.licdn.com/dms/image/C4D0BAQEwg5FK93uumQ/company-logo_200_200/0/1519923012279?e=2147483647&v=beta&t=63CNoS8OTR4lHjPhHSO7eFFqwLGwYunWfyDBV3tdc0c)  
```

I have tested it, and it is showing the same error

![Screenshot 2025-04-11 at 10 57 41 PM](https://github.com/user-attachments/assets/03b73ae7-32d6-45b4-a782-ac9651c948bc)

Reply:
ok, this is weird, I am sure I tested this locally and it worked. But you are right, in the test server it does kind of fail, because if you can notice, it is correctly identifying the file as JPEG. I'll need to dig more to see if there is something in the configs that could tweak the results somehow. See attached vide of how it works locally.

https://github.com/user-attachments/assets/8ef36174-6d95-4e24-bb27-34502b44b842

<desperateVoice>...it works in my laptop...</desperateVoice>

Reply:
ok, found the solution. So, the complete flow for this is that before returning the post, when a post with an image URL is created, the server processes it in post_metadata.go something like:
For each URL, the server:
- Makes GET request to the URL
- Checks the Content-Type header to determine if it's an image
- For images, creates a PostImage object with **format**, dimensions, etc.
So, now that we support the `isLinkInfo` we can check if the format is present, then use the format to determine the `fileType`, and if is not present, then fallback to use the link and let the helper function find the file type from there.

See attached video of the cloud server now working:

https://github.com/user-attachments/assets/602bf128-b35f-40d8-bcf3-da256df500d6

Reply:
Awesome.